File: kattis/cinema2.py
n = [int(i) for i in input().split()][0]
x = [int(i) for i in input().split()]
s = 0
for i in range(len(x)):
    if s + x[i] <= n:
        s += x[i]
    else:
        break
print(len(x) - i)

File: kattis/cinema2.rb
n = gets.chomp.split(' ').map(&:to_i)[0]
x = gets.chomp.split(' ').map(&:to_i)
s = 0
for i in 0...x.length
    if s + x[i] <= n
        s += x[i]
    else
        break
    end
end
puts x.length - i